src/datasets/builder.py

```diff
@@ -880,7 +881,8 @@ def incomplete_dir(dirname):
                     self.info.download_checksums = dl_manager.get_recorded_sizes_checksums()
                     self.info.size_in_bytes = self.info.dataset_size + self.info.download_size
                     # Save info
-                    self._save_info()
+                    if verification_mode is not VerificationMode.NO_CHECKS:
+                        self._save_info()
```

Comment:
We should always save the info. For example it stores the feature types and the arrow shards lengths.

Reply:
hm true...
what should we do then? I don't think it's fine that we get different results running the same command, like:
```python
load_dataset("ds_name")  # error - failed verifications
# then
load_dataset("ds_name", verification_mode="no_checks")  # successfully loaded
# then
load_dataset("ds_name")  # successfully loaded despite it was an error before
```
am I missing something?

Reply:
I don't think we can do anything tbh - e.g. reloading a cached dataset works even if you don't have the downloaded raw files anymore so we can't check for their checksums